### ISSUE TITLE
Please document performance

### ISSUE THREAD

**Issue by meooow25** [j-mie6/range-set#4]
Hi, this seems like a nice library and I would like to use it, but I have no idea what runtime behavior to expect because the documentation makes no mention of time complexities and neither is the internal data structure described anywhere.
Please consider documenting performance characteristics of operations in the Haddocks.

**Comment by j-mie6** [comment]
Absolutely, can (and will) do.

The complexity for each operation is the same as for Data.Set.
This then is faster depending the shape of the data.

**Comment by meooow25** [comment]
> Absolutely, can (and will) do.

Great, thanks!

> The complexity for each operation is the same as for Data.Set.
This then is faster depending the shape of the data.

That's very nice, I'll likely use it then.

**Comment by LyricLy** [comment]
> The complexity for each operation is the same as for Data.Set. This then is faster depending the shape of the data.

Are you sure? It looks like `size` is O(n) instead of O(1) as for Set, for example.

**Comment by j-mie6** [comment]
> > The complexity for each operation is the same as for Data.Set. This then is faster depending the shape of the data.
> 
> Are you sure? It looks like `size` is O(n) instead of O(1) as for Set, for example.

Yes, I'm sure. Although since (original) writing that I did make the size operation O(n) compared to Data.Set (actually looking at the timestamp, I think it was before... The operations `member`, `insert`, `delete`, `union`, `difference`, `intersection` and their derivatives are the same complexities)
